tools/autograd/derivatives.yaml

```diff
@@ -164,13 +164,16 @@
 
 - name: acos(Tensor self) -> Tensor
   self: grad * -((-self * self + 1).rsqrt())
+  result: element_wise
```

Comment:
So I understand what `element_wise` and `linear` are from chatting with you. I don't see any documentation of these around (but I also haven't looked hard enough into the rest of the stack). It would be nice to add documentation for these, especially with an explanation for what they do

Reply:
Yes, I will add an updated version of the issue description here!

Reply:
And in derivatives.yaml